ols/app/models/config.py

```diff
@@ -217,16 +247,29 @@ def __init__(self, data: Optional[dict] = None):
         super().__init__()
         if data is None:
             return
-        self.max_entries = data.get(
-            "max_entries", constants.IN_MEMORY_CACHE_MAX_ENTRIES
-        )
+
+        try:
+            self.max_entries = int(
+                data.get("max_entries", constants.IN_MEMORY_CACHE_MAX_ENTRIES)
+            )
+            if self.max_entries < 0:
+                raise ValueError
+        except ValueError:
+            raise InvalidConfigurationError(
+                "invalid max_entries for memory conversation cache,"
+                " max_entries needs to be a non-negative integer"
+            )
 
     def __eq__(self, other):
         """Compare two objects for equality."""
         if isinstance(other, MemoryConfig):
             return self.max_entries == other.max_entries
         return False
 
+    def validate_yaml(self) -> None:
+        """Validate memory cache config."""
+        pass
```

Comment:
no need for `pass` here as i'm told the comment is sufficient for "implementation"